core/federated/RTI/rti_common.c

```diff
@@ -34,10 +48,18 @@ void initialize_rti_common(rti_common_t* _rti_common) {
 #define IS_IN_CYCLE 2
 
 void invalidate_min_delays_upstream(scheduling_node_t* node) {
-  if (node->min_delays != NULL)
-    free(node->min_delays);
-  node->min_delays = NULL;
-  node->num_min_delays = 0;
+  if (node->all_upstreams != NULL) {
+    free(node->all_upstreams);
+    for (int i = 0; i < rti_common->number_of_scheduling_nodes; i++) {
```

Comment:
Why is this for loop inside the test `node->all_upstreams != NULL`?  Maybe it should be inside a test for `min_delays` being NULL (particularly if you adopt my suggestion above).

Reply:
I renamed `invalidate_min_delays_upstream` to `invalidate_min_delays` and made that function reset delay information of all nodes in [4507043](https://github.com/lf-lang/reactor-c/pull/349/commits/4507043ca782ed8e19901fc58fd5cf5ea06422d9). Also resolved this comment, too.